### ISSUE TITLE
Forced to use lower-case names for the DB tables

### ISSUE THREAD

**Issue by gitfan13** [michaelyaakoby/testfun#18]
Hello Michael,

for whatever reason we started our project with DB tables names IN_CAPITAL_LETTERS. But when we try to execute any unit test, testfun couldn't map the entities to the DB tables.

So we changed the names of all tables in the unit schema to lower_case and kept the UPPER_CASE in the production DB, and it worked well until today...

Now we have a case where want to execute a plain SQL statement, and if the statement has lower_case table names, the unit tests run, but production does not. And if the native SQL contains UPPER_CASE, production works but the unit test fails with

```
javax.persistence.PersistenceException: org.hibernate.exception.SQLGrammarException: could not execute statement
Caused by: org.hibernate.exception.SQLGrammarException: could not execute statement
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Table 'projectU.TABLE_NAME' doesn't exist

```

Any help is appreciated, thanks in advance!

**Comment by michaelyaakoby** [comment]
Hi Boji,

I guess you are running on Linux... with Windows this isn't an issue (see http://stackoverflow.com/questions/6134006/are-table-names-in-mysql-case-sensitive - note that the comment on innodb isn't relevant if you use "file-per-table").

Maybe you can try to ask MySQL to be nice and always use lowercase, see http://cosminswiki.info/index.php/MySQL_case_insensitive_table_and_column_names. I never tried it though.

BTW, in your entities, do you specify table-name?
If you do, Hibernate won't use the "hibernate.ejb.naming_strategy" for table names.

Also, maybe you can change the "hibernate.ejb.naming_strategy" class - the unit-tests of TestFun are using "org.hibernate.cfg.ImprovedNamingStrategy" which lowers everything.

Hope this help...
M.

**Comment by gitfan13** [comment]
Hi Michael,

I appreciate your answer! But it seems like a time-consuming issue to investigate, and I cannot deal with right right now (our go-live is in 2 weeks).

I've implemented this workaround: Since we only have a few DB tables, I've renamed them all to lowercase. Now my unit tests and Hibernate are both happy.

My thoughts and answers to your questions:

- Yes, I'm on Linux
- Yes, I'm annotating my RouteDetails entity with @Table(name = "ROUTE_DETAILS"), etc.
- Even if it's due to mySql issues, I think that testfun's current behavior is a bit unfortunate, especially since hibernate manages to find the UPPER_CASE tables.

But at least the workaround is simple, so I'm closing the issue.  :+
